app/src/ui/repository-settings/fork-settings.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react'
+import { DialogContent } from '../dialog'
+import { ForkContributionTarget } from '../../models/workflow-preferences'
+import { RepositoryWithForkedGitHubRepository } from '../../models/repository'
+
+interface IGitIgnoreProps {
+  readonly forkContributionTarget: ForkContributionTarget
+  readonly repository: RepositoryWithForkedGitHubRepository
+  readonly onForkContributionTargetChanged: (
+    forkContributionTarget: ForkContributionTarget
+  ) => void
+}
+
+enum RadioButtonId {
+  Parent = 'ForkContributionTargetParent',
+  Self = 'ForkContributionTargetSelf',
+}
+
+/** A view for creating or modifying the repository's gitignore file */
+export class ForkSettings extends React.Component<IGitIgnoreProps, {}> {
```

Comment:
```suggestion
export class ForkSettings extends React.Component<IForksSettingProps, {}> {
```

though I bet you that'll need a prettify run